File: practice/algorithms/implementation/find_digits/FindDigits.scala
object FindDigits extends App {
  var inputs = io.Source.stdin.getLines().toList.tail

  inputs.foreach {
    in =>
      println(in.toCharArray.count(d =>
        d.getNumericValue != 0 && in.toInt % d.getNumericValue == 0))
  }
}

File: practice/algorithms/implementation/find_digits/FindDigits.java
import java.util.Scanner;

public class FindDigits 
{
	public static void main(String[] args)
	{
		Scanner stdin = new Scanner(System.in);
		int tests = Integer.parseInt(stdin.nextLine());
		for(int i = 0; i < tests; i++)
		{
			int result = 0;
			String number = stdin.nextLine();
			for(char digit : number.toCharArray())
			{
				if(Character.getNumericValue(digit) != 0 
						&& Integer.parseInt(number) % Character.getNumericValue(digit) == 0)
				{
					result++;
				}
			}
			System.out.println(result);
		}
		stdin.close();
	}
}
